server/src/cg/group4/server/database/query/CreateGroup.java

```diff
@@ -3,57 +3,78 @@
 
 /**
- *
+ * Query used to create a new group.
  */
 public class CreateGroup extends Query {
-
+    /**
+     * The name for the group.
+     */
     protected final String cGroupName;
+    /**
+     * The ownerID for the group.
+     */
     protected final String cOwnerId;
 
-    public CreateGroup(String groupName, String ownerId) {
+    /**
+     * Creates a new CreateGroup Query using the supplied group name and ownerID.
+     * @param groupName The group name.
+     * @param ownerId The owner ID.
+     */
+    public CreateGroup(final String groupName, final String ownerId) {
         cGroupName = groupName;
         cOwnerId = ownerId;
     }
 
-
     @Override
-    public Group query(DatabaseConnection databaseConnection) throws SQLException {
-        Statement statement = databaseConnection.query();
-        statement.executeUpdate("INSERT INTO 'Group' (OwnerId,Name) VALUES ('"
-                + cOwnerId + "', '" + cGroupName + "')");
-        databaseConnection.commit();
+    public Group query(final Connection databaseConnection) throws SQLException {
+        String preparedStatement = "INSERT INTO 'Group' (OwnerId, Name) VALUES (?, ?)";
+
+        Group group;
+
+        try (PreparedStatement statement = databaseConnection.prepareStatement(preparedStatement)) {
+            statement.setString(1, cOwnerId);
+            statement.setString(2, cGroupName);
+            statement.executeUpdate();
+        }
+
+        String preparedQuery = "SELECT G.Key AS GroupId, G.Name AS Name, G.OwnerId As OwnerId, U.Username AS Username, "
+                + "U.Id FROM 'Group' G INNER JOIN User U ON G.OwnerId = U.Id WHERE G.Name = ? AND "
+                + "G.OwnerId = ? LIMIT 1";
+
+        try (PreparedStatement statement = databaseConnection.prepareStatement(preparedQuery)) {
+            statement.setString(1, cGroupName);
+            statement.setString(2, cOwnerId);
 
-        ResultSet resultSet = statement.executeQuery(
-                "SELECT G.Key AS GroupId, G.Name AS Name, G.OwnerId As OwnerId, U.Username AS Username, U.Id "
-                        + "FROM 'Group' G INNER JOIN User U ON G.OwnerId = U.Id "
-                        + "WHERE G.Name = '" + cGroupName + "' AND G.OwnerId = '" + cOwnerId + "' LIMIT 1");
+            GroupData groupData;
 
-        int groupId = resultSet.getInt("GroupId");
+            int groupId;
 
-        GroupData groupData = new GroupData(
-                groupId,
-                resultSet.getString("Name"),
-                resultSet.getString("OwnerId"),
-                resultSet.getString("Username")
-        );
+            try (ResultSet resultSet = statement.executeQuery()) {
+                groupId = resultSet.getInt("GroupId");
 
-        resultSet.close();
-        statement.close();
+                groupData = new GroupData(
+                        groupId,
+                        resultSet.getString("Name"),
+                        resultSet.getString("OwnerId"),
```

Comment:
`resultSet.getString("Name"), resultSet.getString("OwnerId"),` could be replace with `cGroupName, cOwnerId`